.travis.yml

```diff
@@ -5,12 +19,17 @@ rvm:
 - 2.0.0
 
 cache:
-  directories:
-  - $HOME/gcloud/
-  - vendor/bundle
+  - directories:
+    - $HOME/gcloud/
+    - vendor/bundle
 
 env:
-- PATH=$PATH:$HOME/gcloud/google-cloud-sdk/bin GOOGLE_APPLICATION_CREDENTIALS=$TRAVIS_BUILD_DIR/client_secrets.json #Other environment variables on same line
+- TEST_DIR=storage
+  PATH=$PATH:$HOME/gcloud/google-cloud-sdk/bin
+  GOOGLE_APPLICATION_CREDENTIALS=$TRAVIS_BUILD_DIR/client_secrets.json
+- TEST_DIR=pubsub
+  PATH=$PATH:$HOME/gcloud/google-cloud-sdk/bin
+  GOOGLE_APPLICATION_CREDENTIALS=$TRAVIS_BUILD_DIR/client_secrets.json
```

Comment:
Ah, nice I'll do that.

Let's try the matrix, I think I prefer it to some sort of script that loops
through the dirs.

On Tue, Nov 3, 2015 at 3:50 PM, Brent Shaffer <notifications@github.com>
wrote:

> In .travis.yml
> <https://github.com/GoogleCloudPlatform/ruby-docs-samples/pull/5#discussion_r43824907>
> :
>
> >
> >  env:
> > -- PATH=$PATH:$HOME/gcloud/google-cloud-sdk/bin GOOGLE_APPLICATION_CREDENTIALS=$TRAVIS_BUILD_DIR/client_secrets.json #Other environment variables on same line
> > +- TEST_DIR=storage
> > +  PATH=$PATH:$HOME/gcloud/google-cloud-sdk/bin
> > +  GOOGLE_APPLICATION_CREDENTIALS=$TRAVIS_BUILD_DIR/client_secrets.json
> > +- TEST_DIR=pubsub
> > +  PATH=$PATH:$HOME/gcloud/google-cloud-sdk/bin
> > +  GOOGLE_APPLICATION_CREDENTIALS=$TRAVIS_BUILD_DIR/client_secrets.json
>
> While I can't say I'm a huge fan (it'll complicate the build UI by adding
> lots of rows), I'm not hugely against it. However, you should use the
> global keyword for env:
>
> env:
>   global:
>     - PATH=$PATH:$HOME/gcloud/google-cloud-sdk/bin
>     - GOOGLE_APPLICATION_CREDENTIALS=$TRAVIS_BUILD_DIR/client_secrets.json
>   matrix:
>     - TEST_DIR=storage
>     - TEST_DIR=pubsub
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/ruby-docs-samples/pull/5/files#r43824907>
> .
>